code/computational_geometry/src/BresenhamCircleDrawingAlgorithm.cpp

```diff
@@ -0,0 +1,82 @@
+#include <iostream>
+
+class BresenhamCircle
+{
+
+public:
+    BresenhamCircle(int radius_) : radius_(radius_) { }
+
+    void getRadiusCenter();
+
+    void drawBresenhamCircle();
+
+    void displayBresenhmCircle(int xc_, int yc_, int x, int y);
+
+private:
+    int radius_;
+    int xc_;
+    int yc_;
+
+};
+
+void BresenhamCircle::drawBresenhamCircle()
+{
+    int x = 0, y = radius_;
+    int decesionParameter = 3 - 2 * radius_;
+    displayBresenhmCircle(xc_, yc_, x, y);
+    while (y >= x)
+    {
+        x++;
+        if (decesionParameter > 0)
+        {
+            y--;
+            decesionParameter = decesionParameter + 4 * (x - y) + 10;
+        }
+        else
+            decesionParameter = decesionParameter + 4 * x + 6;
+        displayBresenhmCircle(xc_, yc_, x, y); //displaying all the Eight Pixels of (x,y)
+        delay(30);
+    }
+}
+
+void BresenhamCircle::getRadiusCenter()
+{
+    std::cout << "\nEnter Radius of the Circle : ";
+    std::cin >> radius_;
+    std::cout << "\nEnter X-Coordinate of Center of Circle : ";
+    std::cin >> xc_;
+    std::cout << "\nEnter Y-Coordinate of Center of Circle : ";
+    std::cin >> yc_;
+}
+
+void BresenhamCircle::displayBresenhmCircle(int xc_,int yc_, int x, int y)
+{
+    //displaying all 8 coordinates of(x,y) residing in 8-octants
+    putpixel(xc_+x, yc_+y, WHITE);
+    putpixel(xc_-x, yc_+y, WHITE);
+    putpixel(xc_+x, yc_-y, WHITE);
+    putpixel(xc_-x, yc_-y, WHITE);
+    putpixel(xc_+y, yc_+x, WHITE);
+    putpixel(xc_-y, yc_+x, WHITE);
+    putpixel(xc_+y, yc_-x, WHITE);
+    putpixel(xc_-y, yc_-x, WHITE);
+}
+
+int main()
+{
+    int gd = DETECT, gm;
+    initgraph(&gd, &gm, NULL);
```

Comment:
I’m not saying to not use graphics functions I’m saying to include a more modern one in the repository itself. Then you can include this specific file locally. For example you can use SDL2.

Reply:
Sorry as before I was not knowing about SDL, but when I mined about SDL we can just use their inbuilt functions to draw a line, circle and many more and the ALGORITHM OF LINE AND CIRCLE DRAWING which students refer usually is graphics .h when they learn for the very first time. I have to give an implementation of ALGORITHM in C++ for early bird users which they can understand easily.

Reply:
If this is not accepted then will it be acceptable if I write only the function which performs the circle and line drawing which would be then included locally in SDL, or both the files(function template and one including graphics) should I pull the request?

Reply:
Well, to be honest, I'd rather you at least include the `graphics.h` file in the repository itself, just so the code works. The idiomatic alternative would be to use a more modern graphics library. However, if this proves to be too troublesome, then just include `graphics.h` in the repository.

Reply:
Sorry for inconvenient replies as i was not able to figure out the requirement and your instructions sir, i will make the appropriate changes...

Reply:
No, its fine. Ideally we would want to be able to link the file as well, but what you had in your most recent PR was good.

Reply:
To be honest i tried my best to include graphics.h in repository but i don't know why it is not happening as only single file is being included each time you can also check my commits...